src/main/java/com/arpnetworking/metrics/mad/parsers/StatsdToRecordParser.java

```diff
@@ -237,7 +249,7 @@ public StatsdToRecordParser() {
     private static final Splitter LINE_SPLITTER = Splitter.on('\n').omitEmptyStrings();
     private static final ThreadLocal<NumberFormat> NUMBER_FORMAT = ThreadLocal.withInitial(NumberFormat::getInstance);
     private static final Pattern STATSD_PATTERN = Pattern.compile(
-            "^(?<NAME>[^:@|]+):(?<VALUE>[^|]+)\\|(?<TYPE>[^|]+)(\\|@(?<SAMPLERATE>[^|]+))?(\\|#(?<TAGS>.+))?$");
+            "^(?<NAME>[^:@|,]+)(,(?<INFLUXTAGS>[^:@]+))?:(?<VALUE>[^|]+)\\|(?<TYPE>[^|]+)(\\|@(?<SAMPLERATE>[^|]+))?(\\|#(?<TAGS>.+))?$");
```

Comment:
Is there ever a possibility of <string>,:<the rest>, if so then this will not match in that case.

Reply:
I believe this is not allowed and throws a parse exception.  Added test to check it.